### DISCUSSION TITLE
stopPropagation before create or save

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#3033] [Help]
How can I stop the execution of the edit or the create based on a validation within the beforesave or beforedit functions? something like stopPropagation from livewire.

**Comment by danharrin** [comment]
Throw a ValidationException

**Reply by jcgmalave** [reply]
What I don't know is how inside Fialment, to stop the execution of the insert or the update. In this case within the exception.

**Reply by danharrin** [reply]
If you throw a ValidationException, Livewire will halt the script completely and display the error

**Reply by jcgmalave** [reply]
Would it be possible to replace the conventional exception with a Filament NOTIFY?

this:

![imagen](https://user-images.githubusercontent.com/13526704/177375742-2dc287d8-1e09-4797-84b8-8710a04ce772.png)

for this:

![imagen](https://user-images.githubusercontent.com/13526704/177375840-fa68e852-f361-4549-bf87-cd17d3c6363a.png)

thanks a lot, in advanced

**Reply by jcgmalave** [reply]
Even though it throws the exception, the records, once the page is updated, are stored and/or edited

**Reply by danharrin** [reply]
Notify, then throw the exception.

And make sure the exception has at least one message, otherwise it might not be caught

**Comment by studiowizjo** [comment]
Why not just use validation `rules()` on field?

**Reply by jcgmalave** [reply]
Is there any rule to make the field unique, but not within the entire table, but based on a primary key? For example, a record can have many colors, but depending on that record, a color cannot be repeated.

**Reply by studiowizjo** [reply]
You can pass to this method closure and write your own logic. Here is [documentation about this](https://filamentphp.com/docs/2.x/forms/validation#custom-rules) as well to Laravel about this topic.
